RFC-0003-arrow-flight-connector.md

```diff
@@ -0,0 +1,122 @@
+# **RFC0003 for Presto**
+
+## Arrow Flight connector template in Presto
+
+Proposers
+
+* Elbin Pallimalil
+* Ashwin Krishna Kumar
+
+## [Related Issues]
+
+https://github.com/prestodb/presto/issues/22729
+
+## Summary
+
+Create a base Arrow Flight module in Presto which can be extended to create an Arrow Flight connector.
+
+## Background
+
+Apache Arrow Flight is a new general-purpose client-server framework that simplifies high performance transport of large datasets over network interfaces. https://arrow.apache.org/blog/2019/10/13/introducing-arrow-flight/
+
+Presto currently supports different datasources through connectors for each type of datasource. The connectors can use JDBC or some other protocol for datasource connection. If we use Flight client to connect to the different datasources we can improve the performance of the datasource connection. Also we would only need to develop one Flight client that can be used across different datasources instead of developing individual connectors for each type of datasource.
+
+## Proposed Implementation
+
+A template for Flight connector will be created as `presto-base-arrow-flight` module. Since each implementation of flight server will be different, this base module can be extended to create a flight connector that supports a particular implementation. The template will support only read operations.
+
+### High level design
+
+#### Proposed design
+
+![Design diagram](arrow-flight-connector/Presto-flight-client.drawio.png)
+
+The Arrow Flight libraries provide a development framework for implementing a service that can send and receive data streams. Flight APIs are used to retrieve the data from the selected table. The steps below need to be followed. 
+
+1. Create a Flight Client using the host, port, ssl information of the Flight service. 
+    1. Maintain a connection pool of Flight clients.
+    2. Reuse a Flight client from the pool that is already connected to the same location as the new connection.
+    3. Evict connections from the pool using the Least Frequently Used strategy.
+2. Use the Flight Client to connect to the Flight service. 
```

Comment:
May be add point `1.2`about reuse here.

Reply:
Mentioned that the flight client can be re-used from a pool.